cypress/integration/admin/fileManager/uploadImage.spec.js

```diff
@@ -20,18 +68,288 @@ context("File Manager View - CRUD", () => {
+            { fileName: "gifPicture.GIF", type: "image/gif" },
+            { fileName: "sample.jpeg", type: "image/jpeg" }
+        ];
+        files.forEach(({ fileName, type }) => {
+            cy.findByTestId("fm-list-wrapper").dropFile(fileName, type);
+        });
+        cy.findByText("File upload complete.").should("exist");
+
+        // All files should be there.
+        cy.findByTestId("fm-list-wrapper").within(() => {
+            cy.findAllByTestId("fm-list-wrapper-file").should("have.length", files.length);
+        });
+
+        // Assert that only 3 files out of 5 have thumbnail image.
+        cy.get("div.filePreview img").should("have.length", 3);
+
+        cy.visit("/account");
+        cy.get('div[data-role="select-image"]').click();
+
+        // Assert that for avatar we can choose only from 3 files (png/gif/jpeg).
+        cy.get('div[data-testid="fm-list-wrapper-file"]').should("have.length", 3);
+    });
+
+    it("should edit and save file using the flip feature", () => {
+        const fileName = "sample_2.jpeg";
+
+        // Check if there are existing file and delete them.
+        cy.fmListFiles({}).then(files => {
+            for (let i = 0; i < files.length; i++) {
+                deleteFile();
+            }
+        });
+
+        // Drop file
+        cy.findByTestId("fm-list-wrapper").dropFile(fileName, "image/jpeg");
+        cy.findByText("File upload complete.").should("exist");
+
+        // Open file details
+        cy.findByTestId("fm-list-wrapper").within(() => {
+            cy.findAllByTestId("fm-list-wrapper-file")
+                .first()
+                .within(() => {
+                    cy.findByTestId("fm-file-wrapper-file-info-icon").click({ force: true });
+                });
+        });
+
+        // Assert the aside file details is displayed.
+        cy.findByTestId("fm.file-details.drawer").should("be.visible");
+        cy.findByTestId("fm.file-details.drawer").find("img").should("be.visible");
+        cy.get("span").contains("File details").should("be.visible");
+
+        // Edit and save file by using Flip option.
+        cy.findByTestId("fm-edit-image-button").click();
+        cy.findByTestId("fm-image-editor-dialog")
+            .should("be.visible")
+            .within(() => {
+                cy.get('[data-testid="flip-item"]').click();
+                cy.get("span").contains("Apply").click();
+                cy.findByText("Save").click();
+            });
+        cy.contains("File upload complete.").should("be.visible");
+
+        //  Assert that there are displayed both initial item and edited one.
+        cy.get('div[data-testid="fm-list-wrapper-file"]:nth-child(1)')
+            .find("div.label")
+            .contains(fileName);
+        cy.get('div[data-testid="fm-list-wrapper-file"]:nth-child(2)')
+            .find("div.label")
+            .contains(fileName);
+    });
+
+    it("should test minimum and maximum allowed file upload size", () => {
```

Comment:
Can we also revert back the original max file upload size (once the test has finished)?

Reply:
yes, as the last part of the test (the last forEach iteration), we set maximumFileUploadBytesLimit: 26214401 bytes and make sure that we can successfully upload the file as well.